cloud/aws/services/ec2/instances.go

```diff
@@ -0,0 +1,71 @@
+// Copyright © 2018 The Kubernetes Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+// instances is used to scope down/organize the ec2 client.
+type instances interface {
+	DescribeInstances(*ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error)
+	RunInstances(*ec2.RunInstancesInput) (*ec2.Reservation, error)
```

Comment:
Given that these methods are used within an interface, which gives the context, what do you think about dropping the `Instances` suffix at the end? For example [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/55/files#diff-f84006d2b31889959340918e85b391deR44) there is some repetition.

Reply:
I can't change the methods, those are coming from the ec2 client. The goal of this interface is to scope down the client. I'm using a *ec2.EC2 pointer to satisfy this interface I'm defining here. But because i'm defining exactly what it is I need out of the *ec2.EC2 I can mock this in tests by implementing two functions instead of like, 300